openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -12,6 +15,28 @@
 logger = logging.getLogger(__name__)
 
 
+class IngestionError(Exception):
+    """
+    Custom exception which includes information about the query_params that
+    were being used when the error was encountered.
+    """
+
+    def __init__(self, error, traceback, query_params):
+        self.error = error
+        self.traceback = traceback
+        self.query_params = json.dumps(query_params)
+
+    def __str__(self):
+        # Append query_param info to error message
+        return f"""{self.error}
+    query_params: {self.query_params}"""
+
+    def print_with_traceback(self):
+        # Append traceback
+        return f"""{str(self)}
+    {self.traceback}"""
```

Comment:
This will include the traceback, but it looks like it will also include the indentation on the first line. This might be a bit better to prevent having to `dedent` or include erroneous whitespace:
```suggestion
        return f"{str(self)}\n{self.traceback}"
```